packages/astro/src/vite-plugin-astro-server/route.ts

```diff
@@ -273,7 +278,21 @@ export async function handleRoute({
 		});
 	}
 
-	let response = await renderContext.render(mod);
+	let response;
+	try {
+		response = await renderContext.render(mod);
+	} catch (err: any) {
+		const custom500 = getCustom500Route(manifestData);
+		if (!custom500) {
+			throw err;
+		}
+		logger.error('router', err.stack || err.message);
```

Comment:
If I understand correctly, the goal is to have great error logs since the page will probably display less data than the default error overlay?

Reply:
Yes exactly. Your 500 props feature will allow users to better handle the error, but since it's a feature, for now I thought a log could be enough

Reply:
Make sense! Even after the prop introduction, keeping this log makes sense.
```suggestion
		// Log useful information that the custom 500 page may not display unlike the default error overlay
		logger.error('router', err.stack || err.message);
```